core-common/src/main/java/org/glassfish/jersey/message/internal/FormMultivaluedMapProvider.java

```diff
@@ -53,9 +53,12 @@ public FormMultivaluedMapProvider() {
 
     @Override
     public boolean isReadable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-        // Only allow types MultivaluedMap<String, String> and MultivaluedMap.
-        return type == MultivaluedMap.class
-                && (type == genericType || mapType.equals(genericType));
+        if(type == MultivaluedMap.class || mapType.equals(genericType)){
+            return true;
+        }else if("java.util.Map".equals(type.getName()) || "java.util.Map".equals(genericType.getClass().getName())){
+            return true;
+        }
+        return false;
```

Comment:
Should be written as `return {first-if-condition} || {second-if-condition};`.

Also if it is correct to compare `type` with `MultivaluedMap.class` it suggest that doing the same for `Map` should be fine as well. WDYT?

Reply:
I agree, it looks nicer aswell, good catch